### ISSUE TITLE
Leave deletion of secrets to the k8s garbage collector

### ISSUE THREAD

**Issue by awwithro** [bitnami-labs/sealed-secrets#142]
When a SealedSecret object is deleted, the sealed-secrets operator deletes the associated secret. I think this should be left to the k8s garbage-collector. Since the ownerReference field is set, the secret should be garbage collected by k8s anyway.

My particular use case was trying to remove a SealedSecret object but keep the Secret object. I ran the following patch on the secret `k patch secrets my-secret -p '[{"op":"remove","path":"/metadata/ownerReferences"}]' --type=json`. My expectation was that when I deleted the SealedSecret the Secret would remain since the object was now "ownerless". However, the sealed-secrets operator deleted the secret anyway.

**Comment by anguslees** [comment]
Agreed.  The current explicit-deletion behaviour is from a time before server-side garbage collection existed - but it is reasonable to assume this exists now.

**Comment by yann-soubeyrand** [comment]
IMHO it would also be good to have an option to disable deletion of secrets when sealed secrets are deleted. Indeed, if you remove the CRD, all your secrets are gone. Similarly, cert-manager doesn't delete secrets by default when certificates are deleted, you have to add a flag to enable this behaviour: https://github.com/jetstack/cert-manager/pull/819.
